pkg/controllers/webhook/controller.go

```diff
@@ -972,22 +971,28 @@ func addOpnForMutatingWebhookConf(rules []kyvernov1.Rule, mapResourceToOpnType m
 			operationStatusMap = computeOperationsForMutatingWebhookConf(r, operationStatusMap)
 			resources = computeResourcesOfRule(r)
 			for _, r := range resources {
-				mapResourceToOpn, mapResourceToOpnType = appendResource(r, mapResourceToOpn, operationStatusMap, mapResourceToOpnType)
+				gvrss := gvkToGvrs(r, c.discoveryClient, false)
+				for _, gvrs := range gvrss {
+					mapResourceToOpn, mapResourceToOpnType = appendResource(gvrs.Resource, mapResourceToOpn, operationStatusMap, mapResourceToOpnType)
+				}
 			}
 		}
 	}
 	return mapResourceToOpnType
 }
 
-func addOpnForValidatingWebhookConf(rules []kyvernov1.Rule, mapResourceToOpnType map[string][]admissionregistrationv1.OperationType) map[string][]admissionregistrationv1.OperationType {
+func (c *controller) addOpnForValidatingWebhookConf(rules []kyvernov1.Rule, mapResourceToOpnType map[string][]admissionregistrationv1.OperationType) map[string][]admissionregistrationv1.OperationType {
 	var mapResourceToOpn map[string]map[string]bool
 	for _, r := range rules {
 		var resources []string
 		operationStatusMap := getOperationStatusMap()
 		operationStatusMap = computeOperationsForValidatingWebhookConf(r, operationStatusMap)
 		resources = computeResourcesOfRule(r)
 		for _, r := range resources {
-			mapResourceToOpn, mapResourceToOpnType = appendResource(r, mapResourceToOpn, operationStatusMap, mapResourceToOpnType)
+			gvrss := gvkToGvrs(r, c.discoveryClient, false)
```

Comment:
I wonder how it would impact performance if we convert GVKs to GVRs here? Ideally we can create a set of GVKs to reduce client queries while converting to GVRs.